tools/accuracy_checker/accuracy_checker/preprocessor/inpainting_preprocessor.py

```diff
@@ -0,0 +1,171 @@
+        if annotation_meta['mask'].get('mask_name') is not None:
+            warning("Custom mask dataset found but won't be used.")
+
+        img = image.data[0]
+        img_height, img_width = img.shape[:2]
+        mask = np.zeros((img_height, img_width, 1), dtype=np.float32)
+
+        for _ in range(self.parts):
+            mask += self._free_form_mask(self.max_vertex, self.max_length, self.max_brush_width, img_height, img_width)
+        mask = np.minimum(mask, 1.0)
+
+        if len(mask.shape) == 2:
+            mask = np.expand_dims(mask, axis=2)
+
+        img = img * (1 - mask) + 255 * mask
+        image.data = [img, mask]
+        identifier = image.identifier[0]
+        image.identifier = ['{}_image'.format(identifier), '{}_mask'.format(identifier)]
+        return image
+
+
+class RectMask(Preprocessor):
+    __provider__ = "rect_mask"
+
+    @classmethod
+    def parameters(cls):
+        parameters = super().parameters()
+        parameters.update({
+            'mask_height': NumberField(optional=True, default=128, description="Height of mask"),
+            'mask_width': NumberField(optional=True, default=128, description="Width of mask"),
+            'size': NumberField(optional=True, default=128, description="Size of mask, used if both dementions are equal")
+        })
+        return parameters
+
+    def configure(self):
+        self.mask_height, self.mask_width = get_size_from_config(self.config, allow_none=True)
+
+    def process(self, image, annotation_meta=None):
+        if annotation_meta['mask'].get('mask_name') is not None:
+            warning("Custom mask dataset found but won't be used.")
+
+        img = image.data[0]
+        img_height, img_width = img.shape[:2]
+        mp0 = (img_height - self.mask_height)//2
+        mp1 = (img_width - self.mask_width)//2
+
+        mask = np.zeros((img_height, img_width)).astype(np.float32)
+        mask[mp0:mp0 + self.mask_height, mp1:mp1 + self.mask_width] = 1
+        mask = np.expand_dims(mask, axis=2)
+
+        img = img * (1 - mask) + 255 * mask
+        image.data = [img, mask]
+        identifier = image.identifier[0]
+        image.identifier = ['{}_image'.format(identifier), '{}_mask'.format(identifier)]
+        return image
+
+
+class CustomMask(Preprocessor):
+    __provider__ = 'custom_mask'
+
+    @classmethod
+    def parameters(cls):
+        parameters = super().parameters()
+        parameters.update({
+            'inverse_mask': BoolField(optional=True, default=False, description="Inverse mask"),
+            'mask_loader': StringField(optional=True, default='numpy_reader', description="Mask loader")
+        })
+        return parameters
+
+    def configure(self):
+        self.inverse_mask = self.get_value_from_config('inverse_mask')
+        self.mask_loader = self.get_value_from_config('mask_loader')
+
+    def process(self, image, annotation_meta=None):
+        if annotation_meta.get('mask') is None:
```

Comment:
you can make mask_dir preprocessor parameter and do not store it in annotation_meta